src/scipp/testing/strategies.py

```diff
@@ -132,63 +126,59 @@ def variable_args(draw,
                 unique=unique)
 
 
+# This implementation is designed such that the individual strategies
+# for default arguments are constructed only once, namely when
+# `variables` is called. Sampling via `_concrete_args` then reuses
```

Comment:
This implies that when `variables` is called, e.g., for a test case, then the whole generated chain of variables will have, say, the same dtype?

Reply:
Not by default. When you call `variable(dtype=None)`, i.e. default dtype, it is set to the default dtype strategy *when calling `variable`*. But for each example, it draws a new value from that strategy. So you get different dtypes.
The difference compared to before is that it used to create a new strategy *for every example* instead of reusing the same.

The same goes for all arguments.

Reply:
👍